### PR TITLE
Notification: Use @jvalue/node-dry-basics library instead of own implementations

### PR DESCRIPTION
**Pull request by georg-schwarz**

### CODE REVIEW

**Comment by MBuchalik** [comment]
(Not a full review - just something I noticed.) The commit message seems to be broken: It says `Use @value/node-dry-basics` instead of `Use @jvalue/node-dry-basics`. The same also goes for the title of this PR.

**Comment by georg-schwarz** [comment]
Thanks!
It's kind of strange... Linter says ok, but the library cannot be found at runtime. Probably I did something wrong in packaging the lib?

image

**Comment by MBuchalik** [comment]
So it can also be compiled without any issues, but the library is missing during runtime? Or does the compiler also fail at some point?

Are there any logs during runtime, or is the library just `undefined` (or `null`)?

File: notification/package.json [hunk 1]
```diff
@@ -43,7 +43,8 @@
     "jest": "^26.5.3",
     "ts-jest": "^26.4.1",
     "tslib": "^2.0.3",
-    "typescript": "^4.0.3"
+    "typescript": "^4.0.3",
+    "@jvalue/node-dry-basics": "0.0.3"
```
Comment (sonallux) [inline notification/package.json]: You must move this from `devDependencies` to `dependencies`

**Review by sonallux — CHANGES_REQUESTED** [review]

File: notification/package.json (hunk 1 above)
Comment (georg-schwarz) [reply notification/package.json]: oh darn, I'm so stupid... Thanks a lot <3

**Comment by georg-schwarz** [comment]
Ah didn't fix the issue. That's the logs:
image

Does the published lib have to have the index.js at root?

**Comment by MBuchalik** [comment]
Just to make sure: Did you run `npm install` after changing from `devDependencies` to `dependencies`?

**Comment by georg-schwarz** [comment]
Jep :) I rebuilt the docker image without cache, so it should pull in as runtime dependency..

**Comment by sonallux** [comment]
> Jep :) I rebuilt the docker image without cache, so it should pull in as runtime dependency..

@georg-schwarz  in your last commit 2cf851a where you moved the dependency from `devDependencies` to `dependencies` you have not committed the updated `package-lock.json`. This file should have been updated by `npm install`

When I do an `npm install` on your branch locally, rebuild the containers and run the notification integration tests, they all pass 🚀

**Comment by georg-schwarz** [comment]
Phew, that was quite an odyssey... I guess I didn't trust my published npm package :)
Thanks @jsone-studios  and @MBuchalik for your help!

File: notification/src/env.ts [hunk 2]
```diff
@@ -1,27 +1,15 @@
-const isEmpty = (value: string | undefined): value is undefined => value === undefined || value === ''
+import { readEnvOrDie } from '@jvalue/node-dry-basics'
```
Comment (mathiaszinnen) [inline notification/src/env.ts:1]: I really like that name

**Review by mathiaszinnen — APPROVED** [review]
Nice!

**Pull request merged**